projects/packages/image-cdn/src/class-image-cdn.php

```diff
@@ -0,0 +1,1386 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Class for photon functionality.
+ *
+ * @package automattic/jetpack-image-cdn
+ */
+
+namespace Automattic\Jetpack\Image_CDN;
+
+/**
+ * Class Image_CDN
+ */
+class Image_CDN {
+
+	const PACKAGE_VERSION = '0.1.0-alpha';
+
+	/**
+	 * Singleton.
+	 *
+	 * @var null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Allowed extensions.
+	 *
+	 * @var string[] Allowed extensions must match https://code.trac.wordpress.org/browser/photon/index.php#L41
+	 */
+	protected static $extensions = array(
+		'gif',
+		'jpg',
+		'jpeg',
+		'png',
+		'webp', // Jetpack assumes Photon_OpenCV backend class is being used on the server. See link in docblock.
+	);
+
+	/**
+	 * Image sizes.
+	 *
+	 * Don't access this directly. Instead, use self::image_sizes() so it's actually populated with something.
+	 *
+	 * @var array Image sizes.
+	 */
+	protected static $image_sizes = null;
+
+	/**
+	 * Singleton implementation
+	 *
+	 * @return object
+	 */
+	public static function instance() {
+		if ( ! is_a( self::$instance, 'Automattic\Jetpack\Image_CDN' ) ) {
+			self::$instance = new Image_CDN();
```

Comment:
Nitpick 🙂
```suggestion
		if ( ! is_a( self::$instance, self::class ) ) {
			self::$instance = new self();
```
(or `static` instead of `self`)